### DISCUSSION TITLE
when needing accumulative data | contextual data

### DISCUSSION THREAD

**Discussion by speedfl** [jonaskello/eslint-plugin-functional#322] [Q&A] [answered]
Hello,

Some time you need to accumulate data.

Example:

```
export const exec: execCommand = (command: string, args: ReadonlyArray<string> | null): Promise<string[]> => {
    return new Promise<string[]>((resolve, reject) => {

        const data: string[] = [];

        const execution = spawn(command, args);

        execution.on('close', () => {
            resolve(data);
        });

        execution.stdout.on('data', (buffer) => {
            const message = new StringDecoder('utf-8').write(buffer);
            data.push(message);
        });

        execution.stderr.on('data', (buffer) => {
            const message = new StringDecoder('utf-8').write(buffer);
            data.push(message);
        });

        execution.on('error', (error) => {
            reject(error)
        });
    });
```

We will have two issues:

```
Only readonly arrays allowed.eslintfunctional/prefer-readonly-type
Modifying an array is not allowed.eslintfunctional/immutable-data
```

What do you suggest ?

**Comment by RebeccaStevens** [comment, marked as answer]
There are few ways to solve this.

Option one is to explicitly mark the array as mutable (usually by prefixing "mutable" to the start of the variables name) and configure the rules to ignore variable marked like this.

Option two would be to configure the rules to ignore all local mutations.

Option 3 would be to use an eslint-disable comment at the start of the function, and an eslint-enable comment at the end (for the rules being violated).

**Reply by speedfl** [reply]
Thanks a lot @RebeccaStevens.

I initially put eslint-disabled but I like the idea of mutable!
